demos/nbody/sim.hpp

```diff
@@ -294,16 +296,24 @@ class GravSim {
                   return G * acc;
                 };
 
+                vec3<num_t> wvelTmp;
```

Comment:
I don't understand the purpose of this change

Reply:
Because of this: 

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.26.28801\include\tuple(163,43): error GB19FFA8B: binding value of type '__global vec<...>' to reference to type 'vec<...>' drops <<ERROR>> qualifiers
constexpr _Tuple_val(_Other&& _Arg) : _Val(_STD forward<_Other>(_Arg)) {}
```

Reply:
OK that's wild. Should `__global` be showing up on the host compiler?

Reply:
this is not the host compiler. It's compute++. The expression `wvel[i]` is a reference to `__global` and this is incompatible with the return type of the `integrate` function